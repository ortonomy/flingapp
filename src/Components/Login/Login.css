.Login {
    height: 100vh;
    width: 100vw;
    background: linear-gradient(to bottom, rgba(255, 221, 128, 1) 0%, rgba(251, 199, 57, 1) 100%);
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
}

.Login .login-box {
    height: 65%;
    width: 400px;
    background: rgba(238,238,238,0.65);
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: space-around;
}

.Login .login-image {
    flex: 2;
    display: flex;
    flex-flow: column nowrap;
    align-items: space-around;
    justify-content: center;
}

.Login .LoginNav {
    flex: 5;
}

.Login .login-footer {
    flex:1;
    color: #4A90E2;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
}

.Login .login-footer * {
    padding: 0em 0.3em;
}

.Login .tabs {
    height: 2.5em;
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: center;
}

.Login .tabs .tab {
    width: 50%;
    height: 100%;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    color: #4A90E2;
}

.Login .tabs .tab.active {
    background: rgba(255,255,255,1);
    color: #555;
}

.Login .login-body {
    width: 100%;
    height: 85%;
    background: rgba(255,255,255,1);
}
