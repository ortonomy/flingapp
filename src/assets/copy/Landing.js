export const Hero = {
    heroTitle1: "Freelancer",
    heroTitle2: "Management.",
    heroTitle3: "For free.",
    heroSubTitle1: "Keep track of your freelancers' performance and identify the best resources for your next project.",
    heroButtonText1: " Get started "
};