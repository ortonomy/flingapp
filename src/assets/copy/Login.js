export const LoginFooter = {
    link1: {
      text: 'Terms & conditions',
      route: '/terms'
    },
    link2: {
      text: 'Privacy policy',
      route: '/privacy'
    }
};