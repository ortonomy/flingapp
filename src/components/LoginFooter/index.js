// basic react
import React from 'react';

// dependent components
import { Link } from 'react-router-dom';

// styles
import styles from './style.module.css';

const LoginFooter = (props) => {
  return (
    <div className={styles.Footer}>
      {
        props.links.map( (link, key) => {
          return (
            <Link key={key} to={link.route}>
              <span>&nbsp;{link.text}&nbsp;</span>{ key < props.links.length - 1 ? (<span>&nbsp;|&nbsp;</span>) : '' }
            </Link>
          )
        })
      }
    </div>
  );
}

export default LoginFooter