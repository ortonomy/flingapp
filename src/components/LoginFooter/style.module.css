.Footer {
    color: #4A90E2;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
    font-size: 1.2rem;
    margin: 0 0 30px 0;
}