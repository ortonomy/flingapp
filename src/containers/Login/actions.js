// action type constants
export const LOGIN = 'LOGIN';


// all action types
export const actionTypes = {
  LOGIN
};

// action creators
export const login = () => ({
  type: LOGIN
});

// export const appUpdate = value => ({
//   type: APP_UPDATE, 
//   payload: value 
// });

// export const appUpdateSuccess = result => ({
//   type: APP_UPDATE_SUCCESS, 
//   payload: result
// });

// export const appUpdateFail = err => ({
//   type: APP_UPDATE_FAIL, 
//   payload: err, 
//   error: true
// });

// all action creators
export const actions = {
  login
}

