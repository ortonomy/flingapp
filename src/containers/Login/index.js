export { default as Login } from './component';
export { default as logic } from './logic';
export { default as reducer } from './reducer';