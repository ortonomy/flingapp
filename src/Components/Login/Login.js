import React, { Component } from 'react';
import './Login.css';
import LoginForm from '../LoginForm/LoginForm';
import logo from '../../assets/images/full_title_compact_trans.png';

class Login extends Component {
    render() {
        return (
            <div className="Login">
                <div className="login-box">
                    <img className="login-image" src={logo} alt="Fling" />
                    <LoginForm />
                    <div className="login-footer">
                        <a href="#">Terms and Conditions</a>
                        |
                        <a href="#">Privacy Policy</a>
                    </div>
                </div>
            </div>
        )
    }
}

export default Login;
