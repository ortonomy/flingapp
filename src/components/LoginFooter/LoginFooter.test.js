import React from 'react';
import LoginFooter from '../LoginFooter';

// Link information from /assets
import { LoginFooter as LoginFooterCopy } from '../../assets/copy/Login';

it('renders without crashing', () => {
  const wrapper = shallow(<LoginFooter links={[LoginFooterCopy.link1,LoginFooterCopy.link2]} />);
  expect(wrapper).toMatchSnapshot();
});
