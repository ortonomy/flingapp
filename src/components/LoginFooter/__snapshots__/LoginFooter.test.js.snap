// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginFooter
    links={
        Array [
            Object {
              "route": "/terms",
              "text": "Terms & conditions",
            },
            Object {
              "route": "/privacy",
              "text": "Privacy policy",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/terms"
>
          <span>
                     
                    Terms & conditions
                     
          </span>
          <span>
                     | 
          </span>
</Link>,
        <Link
          replace={false}
          to="/privacy"
>
          <span>
                     
                    Privacy policy
                     
          </span>
          
</Link>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <span>
               
              Terms & conditions
               
</span>,
            <span>
               | 
</span>,
          ],
          "replace": false,
          "to": "/terms",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Terms & conditions",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Terms & conditions",
              " ",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " | ",
            },
            "ref": null,
            "rendered": " | ",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <span>
               
              Privacy policy
               
</span>,
            "",
          ],
          "replace": false,
          "to": "/privacy",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Privacy policy",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Privacy policy",
              " ",
            ],
            "type": "span",
          },
          "",
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/terms"
>
            <span>
                         
                        Terms & conditions
                         
            </span>
            <span>
                         | 
            </span>
</Link>,
          <Link
            replace={false}
            to="/privacy"
>
            <span>
                         
                        Privacy policy
                         
            </span>
            
</Link>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span>
                 
                Terms & conditions
                 
</span>,
              <span>
                 | 
</span>,
            ],
            "replace": false,
            "to": "/terms",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Terms & conditions",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Terms & conditions",
                " ",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " | ",
              },
              "ref": null,
              "rendered": " | ",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span>
                 
                Privacy policy
                 
</span>,
              "",
            ],
            "replace": false,
            "to": "/privacy",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Privacy policy",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Privacy policy",
                " ",
              ],
              "type": "span",
            },
            "",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
